crates/ethereum/payload/src/lib.rs

```diff
@@ -251,19 +250,19 @@ where
 
         // There's only limited amount of blob space available per block, so we need to check if
         // the EIP-4844 can still fit in the block
-        if let Some(blob_tx) = tx.as_eip4844() {
-            let tx_blob_gas = blob_tx.blob_gas();
-            if sum_blob_gas_used + tx_blob_gas > MAX_DATA_GAS_PER_BLOCK {
+        if let Some(_blob_tx) = tx.as_eip4844() {
+            blob_txn_count += 1;
```

Comment:
this is a bit more tricky.

we also need a temporary variable here.

the blob count of the tx is blob_tx.versioned_hashes().len()